src/commands/deploy.ts

```diff
@@ -18,16 +18,18 @@ export abstract class DeployCommand extends BaseCommand {
 
   static flags = {
     ...BaseCommand.flags,
-    auto_approve: Flags.boolean({
+    auto_approve: booleanString({
       exclusive: ['compose-file', 'compose_file'],
       description: `${BaseCommand.DEPRECATED} Please use --auto-approve.`,
       hidden: true,
       sensitive: false,
+      default: undefined,
```

Comment:
why is this undefined?

Reply:
booleanString requires a default fallback value to support --flag_name usage, triggering the deprecated messages even if the user had not passed the deprecated flag (failing tests).  Because the deprecated flag has been directly replaced, we set the default value (false) on the active flag and undefined on the deprecated command.  This way,  it does not trigger the deprecated warning unless --auto_approve is specified and present in argv. 

- if neither deprecated or active flag are specified, the active flag will assume the default value (false). 
- if deprecated flag alone is specified, the active flag is assumed the deprecated value and a warning is produced.
- if active flag alone is specified, the active flag will be assigned the passed value.
- if both flags are specified, the active flag will assume the value passed to the active flag (added a check in next commit)